thunderfish/eodanalysis.py

```diff
@@ -0,0 +1,174 @@
+    is calculated.
+
+    :param eod_data:            (2-D array) An array of lists. Each list contains the data around a EOD.
+    :return mean_eod:           (1-D array) Array of the mean data calculated from the 2-D eod_data array.
+    :return std_eod:            (1-D array) Array of the std data calculated from the 2-D eod_data array.S
+    """
+    mean_eod = np.zeros(len(eod_data[0]))
+    std_eod = np.zeros(len(eod_data[0]))
+
+    for i in np.arange(len(eod_data[0])):
+        mean_eod[i] = np.mean([eod_data[eod][i] for eod in np.arange(len(eod_data))])
+        std_eod[i] = np.std([eod_data[eod][i] for eod in np.arange(len(eod_data))], ddof=1)
+
+    return mean_eod, std_eod
+
+def eodanalysisplot(time, mean_eod, std_eod, ax):
+    """
+    Creates a axis for plotting  to visualize a mean eod and its standard deviation.
+
+    :param time:                (1-D array) containing the time of the mean eod.
+    :param mean_eod:            (1-D array) containing the data of the mean eod.
+    :param std_eod:             (1-D array) containing the data of the std eod.
+    :param ax:                  (axis for plot) empty axis that is filled with content in the function.
+    :return:                    (axis for plot) axis that is ready for plotting explaining what the modul did.
+    """
+    u_std = [mean_eod[i] + std_eod[i] for i in np.arange(len(mean_eod))]
+    l_std = [mean_eod[i] - std_eod[i] for i in np.arange(len(mean_eod))]
+    ax.plot(time, mean_eod, lw=2, color='firebrick', alpha=0.7, label='mean EOD')
+    ax.fill_between(time, u_std, l_std, color='grey', alpha=0.3)
+    ax.set_xlabel('time [sec]')
+    ax.set_ylabel('Amplitude (mV)')
+    ax.set_xlim([min(time), max(time)])
+    return ax
+
+def load_example_data(audio_file, channel=0, verbose=None):
+    """
+    This function shows in part the same components of the thunderfish.py poject. Here several moduls are used to load
+    some data to dispay the functionality of the eodanalysis.py modul.
+
+    :param audio_file:      (string) filepath of a audiofile that shall be used for the analysis.
+    :return power:          (1-D array) power array of a psd.
+    :return freqs:          (1-D array) frequency array of a psd.
+    """
+    cfg = ct.get_config_dict()
+
+    if verbose is not None:
+        cfg['verboseLevel'][0] = verbose
+    channel = channel
+
+    # load data using dataloader module
+    print('loading example data ...\n')
+    data, samplrate, unit = dl.load_data(audio_file)
+
+    # calculate best_window
+    print('calculating best window ...\n')
+    bwin_data, clip = bw.best_window(data, samplrate)
+
+    print('calculating powerspectrum ...\n')
+    power, freqs = ps.powerspectrum(data, samplrate)
+
+
+    psd_type, proportion = pt.psd_assignment(power, freqs)
+
+    return bwin_data, samplrate, psd_type
+
+def eod_analysis(bwin_data, samplerate, fish_type, psd_type, plot_data_func=None, **kwargs):
```

Comment:
The eod_analysis function takes an array of data as input and is the main function of the module. The eod_mean function takes only the data around detected eods as input (list of lists) and calulates the mean and std of all detected eods. So I would not combine them.